ddon-90/demo.update-price-api-wrapper
# feat: use auth from headers request

#### What is the purpose of this pull request?

Use request's headers like auth for VTEX internal APIs

#### What problem is this solving?

Use request's headers like auth for VTEX internal APIs

#### How to test it?

The operation should not change:

- Use a valid api key and app token to change the price of a product given its RefId, as a result the price should change.

- Use a NOT valid api key and app token to change the price of a given product's RefId, as a result an authentication error should appear in the API response.

#### Screenshots or example usage

`PUT https://{{workspace}}--{{account}}.myvtex.com/_v/api/pricing/prices/{{RefId}}`
data:

``` json
{
    "markup": 0,
    "basePrice": {{price}},
    "fixedPrices": []
}
```
headers: 
```json
{
    "X-VTEX-API-AppKey": "{{X-VTEX-API-AppKey}}",
    "X-VTEX-API-AppToken": "{{X-VTEX-API-AppToken}}"
}
```

#### How does this PR make you feel? [:link:](http://giphy.com/)

![](https://media2.giphy.com/media/33OrjzUFwkwEg/giphy.gif?cid=ecf05e471xw58gsip9udngr1mr4e4h0dalnpir9j2dd2eykn&ep=v1_gifs_search&rid=giphy.gif&ct=g)

#### Types of changes
